Panic runtime error for 0.4.0 in GKE 1.5.2

Hello,

I just tried to upgrade kube-state-metric pod in 0.4.0. I'm running a GKE cluster in 1.5.2.
Kube-state-metrics is running in Ubuntu 16.04 container and built using:
```
apt-get install -y golang git
cd /root
mkdir -p "$GOPATH"
go get github.com/kubernetes/kube-state-metrics
mv "${GOPATH}/bin/kube-state-metrics" /usr/local/bin/
```


Here is the stacktrace I got when I try to get /metric

```
kube-state-metrics
I0210 07:30:30.326926     175 main.go:139] Using default collectors
I0210 07:30:30.327196     175 main.go:186] service account token present: true
I0210 07:30:30.327224     175 main.go:187] service host: https://10.255.240.1:443
I0210 07:30:30.327902     175 main.go:213] Testing communication with server
I0210 07:30:30.359162     175 main.go:218] Communication with server successful
I0210 07:30:30.359497     175 main.go:263] Active collectors: resourcequotas,replicasets,daemonsets,deployments,pods,nodes
I0210 07:30:30.359533     175 main.go:227] Starting metrics server: :80
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x402578]
goroutine 81 [running]:
panic(0x18c3a00, 0xc82000a0a0)
	/usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
main.(*deploymentCollector).collectDeployment(0xc820456490, 0xc82006e900, 0x0, 0x0, 0x0, 0x0, 0xc820495bc4, 0x6, 0x0, 0x0, ...)
	/root/go/src/github.com/kubernetes/kube-state-metrics/deployment.go:153 +0x238
main.(*deploymentCollector).Collect(0xc820456490, 0xc82006e900)
	/root/go/src/github.com/kubernetes/kube-state-metrics/deployment.go:135 +0x232
github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc8204842c0, 0xc82006e900, 0x7f4865a756a8, 0xc820456490)
	/root/go/src/github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x58
created by github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/root/go/src/github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x360
```

Hello @brancz ,

I have same behavior with you image:
```
I0210 16:30:55.774545       1 main.go:139] Using default collectors
I0210 16:30:55.774749       1 main.go:186] service account token present: true
I0210 16:30:55.774758       1 main.go:187] service host: https://10.255.240.1:443
I0210 16:30:55.775172       1 main.go:213] Testing communication with server
I0210 16:30:55.802572       1 main.go:218] Communication with server successful
I0210 16:30:55.802770       1 main.go:263] Active collectors: resourcequotas,replicasets,daemonsets,deployments,pods,nodes
I0210 16:30:55.802784       1 main.go:227] Starting metrics server: :8080
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x40218e]
goroutine 126 [running]:
panic(0x13249e0, 0xc420014030)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
main.(*deploymentCollector).collectDeployment(0xc42047a3a0, 0xc4202bf800, 0x0, 0x0, 0x0, 0x0, 0xc420257454, 0x6, 0x0, 0x0, ...)
	/home/brancz/go/src/k8s.io/kube-state-metrics/deployment.go:153 +0x27e
main.(*deploymentCollector).Collect(0xc42047a3a0, 0xc4202bf800)
	/home/brancz/go/src/k8s.io/kube-state-metrics/deployment.go:135 +0x133
k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc4209fdbd0, 0xc4202bf800, 0x1c77b80, 0xc42047a3a0)
	/home/brancz/go/src/k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x63
created by k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/home/brancz/go/src/k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x326
```

--- a/deployment.go
+++ b/deployment.go
@@ -150,6 +150,10 @@ func (dc *deploymentCollector) collectDeployment(ch chan<- prometheus.Metric, d 
 	addGauge(descDeploymentSpecReplicas, float64(*d.Spec.Replicas))
 	addGauge(descDeploymentMetadataGeneration, float64(d.ObjectMeta.Generation))
 
+	if d.Spec.Strategy.RollingUpdate == nil {
+		return
+	}
+
 	maxUnavailable, err := intstr.GetValueFromIntOrPercent(d.Spec.Strategy.RollingUpdate.MaxUnavailable, int(*d.Spec.Replicas), true)
 	if err != nil {
 		glog.Errorf("Error converting RollingUpdate MaxUnavailable to int: %s", err)

--- a/deployment_test.go
+++ b/deployment_test.go
@@ -115,11 +115,6 @@ func TestDeploymentCollector(t *testing.T) {
 					Spec: v1beta1.DeploymentSpec{
 						Paused:   true,
 						Replicas: &depl2Replicas,
-						Strategy: v1beta1.DeploymentStrategy{
-							RollingUpdate: &v1beta1.RollingUpdateDeployment{
-								MaxUnavailable: &depl2MaxUnavailable,
-							},
-						},
 					},
 				},
 			},
@@ -130,6 +125,9 @@ func TestDeploymentCollector(t *testing.T) {
 				kube_deployment_spec_paused{namespace="ns2",deployment="depl2"} 1
 				kube_deployment_spec_replicas{namespace="ns1",deployment="depl1"} 200
 				kube_deployment_spec_replicas{namespace="ns2",deployment="depl2"} 5
+				kube_deployment_spec_strategy_rollingupdate_max_unavailable{deployment="depl1",namespace="ns1"} 10
+				kube_deployment_status_observed_generation{namespace="ns1",deployment="depl1"} 111
+				kube_deployment_status_observed_generation{namespace="ns2",deployment="depl2"} 1111
 				kube_deployment_status_replicas{namespace="ns1",deployment="depl1"} 15
 				kube_deployment_status_replicas{namespace="ns2",deployment="depl2"} 10
 				kube_deployment_status_replicas_available{namespace="ns1",deployment="depl1"} 10
@@ -138,10 +136,6 @@ func TestDeploymentCollector(t *testing.T) {
 				kube_deployment_status_replicas_unavailable{namespace="ns2",deployment="depl2"} 0
 				kube_deployment_status_replicas_updated{namespace="ns1",deployment="depl1"} 2
 				kube_deployment_status_replicas_updated{namespace="ns2",deployment="depl2"} 1
-				kube_deployment_status_observed_generation{namespace="ns1",deployment="depl1"} 111
-				kube_deployment_status_observed_generation{namespace="ns2",deployment="depl2"} 1111
-				kube_deployment_spec_strategy_rollingupdate_max_unavailable{namespace="ns1",deployment="depl1"} 10
-				kube_deployment_spec_strategy_rollingupdate_max_unavailable{namespace="ns2",deployment="depl2"} 1
 			`,
 		},
 	}

